dj/typing.py

```diff
@@ -227,20 +249,49 @@ def __getitem__(self, keys) -> "ColumnType":
 
     @classmethod
     def _validate_type(cls, type_: str):
-        test = TYPE_PATTERN.match(type_)
-        if test is not None:
-            outer = test.group("outer")
-            inner = test.group("inner")
-            if outer not in COMPLEX_TYPES:
-                raise ColumnTypeError(f"{outer} is not a KNOWN complex type.")
-            inners = inner.split(",") if outer == ColumnType.MAP else inner
-            return ColumnType(outer)[inners]
+        nested = parse_type(type_)[0]
+        if isinstance(nested, list) and len(nested) > 1:
+            if nested[0] not in COMPLEX_TYPES:
+                raise ColumnTypeError(f"{nested[0]} is not a KNOWN complex type.")
+            return ColumnType(nested[0])[nested[1]]
         type_ = ALIASES.get(type_, type_)
         if type_ not in PRIMITIVE_TYPES:
             raise ColumnTypeError(f"{type_} is not an acceptable type.")
         return ColumnType(type_)
 
 
+def parse_type(expr: str) -> List:
+    """
+    Parses a nested type expression.
+    """
+    tokenizer = re.compile(r"\s*([\[\]\,])\s*").split
+    tokens = list(filter(None, tokenizer(expr)))
+
+    stack = collections.deque()  # type: ignore
+    top = items = []  # type: ignore
+
+    for i, token in enumerate(tokens):
+        next_token = tokens[i + 1] if i + 1 < len(tokens) else None
+        if token == "[":
+            stack.append(items)
+            items.append([])
+            items = items[-1]
+        elif token == "]":
+            if not stack:
+                raise ColumnTypeError("Unbalanced parentheses")
```

Comment:
Would it be easy to include more information here in this error message, maybe even just dump the expression in?
`ColumnTypeError(f"Unbalanced parentheses: {expr}")`

Reply:
Makes sense, I'll dump the expression in